lib/src/backend.rs

```diff
@@ -125,11 +125,43 @@ impl MergedTreeId {
     }
 }
 
+/// An optionally versioned copy source.
+#[derive(ContentHash, Debug, PartialEq, Eq, Clone)]
+pub struct CopySource {
+    /// The path the target was copied from.
+    pub path: RepoPathBuf,
+    /// The specific version the target was copied from. If unspecified, the
+    /// copy comes from the parent commit of the target file version.
+    pub commit_id: Option<CommitId>,
+}
+
+/// A singular copy event in a specific file's history.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct CopyTrace {
+    /// The file that was copied into.
+    pub target: RepoPathBuf,
+    /// The commit where the file was copied.
+    pub commit_id: CommitId,
+    /// The source of the copy.
+    pub source: CopySource,
+}
+
+/// Map from target -> source
+///
+/// This can be set on write to explicitly record copies in the backend. The
+/// backend may discard these silently if it does not support explicit copy
+/// tracking (e.g. git). Backends which do support explicit copy tracking may
```

Comment:
This is the only place you use "copy tracking". Everywhere else is "copy trace" or "copy tracing".
You could make the purpose clearer by using "file rename", if that's what it's for.

Reply:
So the distinction here is somewhat intentional.  I rephrased it a bit to make it clearer.

In my mind, 'copy tracing' is answering the question "where was this file copied from?" which git *does* support in a limited capacity by inferring copies of similar content between two tree states.  Copy *tracking* is the explicit record of "Hey vcs, remember that I copied foo.cc -> bar.cc here" which git does *not* support.